iidm/iidm-tck/src/test/java/com/powsybl/iidm/network/tck/AbstractNodeBreakerInternalConnectionsTest.java

```diff
@@ -220,43 +219,31 @@ private InternalConnections findInternalConnectionsTraverseStoppingAtTerminals(V
         InternalConnections cs = new InternalConnections();
 
         VoltageLevel.NodeBreakerView topo = vl.getNodeBreakerView();
-        int[] nodes = topo.getNodes();
-        final TIntSet explored = new TIntHashSet();
-        for (int n : nodes) {
-            if (explored.contains(n) || topo.getTerminal(n) == null) {
-                continue;
-            }
-            explored.add(n);
-            topo.traverse(n, (n1, sw, n2) -> {
-                explored.add(n2);
+        topo.traverse(topo.getNodes(), (n1, sw, n2) -> {
+            if (topo.getTerminal(n2) == null) {
                 if (sw == null) {
                     cs.add(n1, n2);
                 }
-                return topo.getTerminal(n2) == null ? TraverseResult.CONTINUE : TraverseResult.TERMINATE;
-            });
-        }
+                return TraverseResult.CONTINUE;
+            } else {
+                return TraverseResult.TERMINATE;
+            }
+        });
+
```

Comment:
@zamarrenolm I changed a bit the stop criteria here, hence the test assert, but from what I understood you were only testing the traversal of internal connections with a random stop criteria? then my changes would be ok - otherwise I'd revert that part.